src/main/java/com/faforever/client/game/CustomGamesController.java

```diff
@@ -122,15 +123,32 @@ public TilesSortingOrder fromString(String string) {
     eventBus.register(this);
   }
 
-  public void onShowPrivateGames(ActionEvent actionEvent) {
-    CheckBox checkBox = (CheckBox) actionEvent.getSource();
-    boolean selected = checkBox.isSelected();
-    if (selected) {
-      filteredItems.setPredicate(OPEN_CUSTOM_GAMES_PREDICATE);
-    } else {
-      filteredItems.setPredicate(OPEN_CUSTOM_GAMES_PREDICATE.and(gameInfoBean -> !gameInfoBean.getPasswordProtected()));
+    public void onShowPrivateGames(ActionEvent actionEvent) {
```

Comment:
You should save the state of the Checkbox in the preferences. Cosider binding the checkedProperty of the CheckBox to the property in the preferences. See https://github.com/FAForever/downlords-faf-client/blob/1bee186d0a576fd1dc00cdc10874bc4599f666db/src/main/java/com/faforever/client/preferences/ui/SettingsController.java#L168